### PR TITLE
HPCC-22357 ESDL WSDL scapps compatibility

### PR DESCRIPTION
**Pull request by asselitx**
Change ESDL generated WSDLs to more closely match those generated by scapps.
This aids in our migration of scapps services to DESDL.

 - Output xml declaration
 - Set root of soapAction to service name
 - Remove annotation elements used only for rendering web form
 - Remove unused string element

Signed-off-by: Terrence Asselin <terrence.asselin@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [x] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [x] I have checked that this change does not introduce memory leaks.
  - [x] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by asselitx** [comment]
@timothyklemm Could you please review this change?

File: esp/services/esdl_svc_engine/esdl_binding.cpp [hunk 1]
```diff
@@ -1959,13 +1959,11 @@ void EsdlBindingImpl::initEsdlServiceInfo(IEsdlDefService &srvdef)
     //setWsdlAddress(bndcfg->queryProp("@wsdlServiceAddress"));
 
     IProperties *xsdparams = createProperties(false);
-    xsdparams->setProp( "all_annot_Param", 1 );
```
Comment (timothyklemm) [inline esp/services/esdl_svc_engine/esdl_binding.cpp:1962]: Is there a possible code path on which this property (and the one below) is referenced? End users don't need our own extra annotations, but has it been confirmed that we don't?

Comment (asselitx) [reply esp/services/esdl_svc_engine/esdl_binding.cpp:1962]: @timothyklemm Yes you were correct. I've pushed up a fix to change the parameter value only when the transaction is an explicit request for the XSD or WSDL. Other uses (such as to generate the web form) will use the default XSL parameters set as above. Please re-review.

**Review by timothyklemm — APPROVED** [review]

**Comment by richardkchapman** [comment]
@afishbeck Can you review too please. Also, please squash

File: esp/esdllib/esdl_def.cpp [hunk 2]
```diff
@@ -1689,6 +1689,40 @@ void EsdlDefinition::walkDefinitionDepthFirst( AddedObjs& foundByName, EsdlDefOb
...
+    //    - 'flags' parameter indicates if the caller wants ArrayOf
+    //      entries generated
+    //    - Finally, only EsdlTypeStruct objects used as array items need
+    //      ArrayOf element definitions.
+    if (found && (state & DEPFLAG_ARRAYOF) && (flags & DEPFLAG_ARRAYOF) && EsdlTypeStruct == esdlType)
+    {
+        ForEachItemIn(i, dependencies)
+        {
+            EsdlDefObjectWrapper* wrapper=dependencies[i];
+            if (strcmp(wrapper->queryObject().queryName(), name) == 0)
```
Comment (afishbeck) [inline esp/esdllib/esdl_def.cpp:1719]: Trivial, but prefer the streq() function to strcmp()==0 these days.

**Review by afishbeck — APPROVED** [review]
@asselitx other than a trivial comment looks good.

**Pull request merged**